### PR TITLE
ServiceNow BYOI

### PR DESCRIPTION
**Pull request by gshriki**
ServiceNow integration in BYOI (JS)

Incidents API:
- servicenow-incidents-get-incident - Retrieve ticket information by specific ticket ID
- servicenow-incident-create - Create new ServiceNow ticket
- servicenow-incident-update - Update specific ticket by providing ticket id
- servicenow-incident-add-link - Add link to specific ticket by providing ticket id
- servicenow-incidents-query - Retrieve info with query
- servicenow-incident-upload-file - Upload new file
- fetching incidents

### CODE REVIEW

File: Integrations/integration-ServiceNow.yml [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+description: IT service management
+configuration:
```
Comment (aviadl) [inline Integrations/integration-ServiceNow.yml]: Seems that we are missing API version param
And also insecure

Comment (gshriki) [inline Integrations/integration-ServiceNow.yml]: added both api version and insecure.

File: Integrations/integration-ServiceNow.yml [hunk 2]
```diff
@@ -0,0 +1,802 @@
+commonfields:
+  id: ServiceNow
+  version: -1
```
Comment (Spivaka) [inline Integrations/integration-ServiceNow.yml]: no need for fromversion here?

File: Integrations/integration-ServiceNow.yml [hunk 3]
```diff
@@ -0,0 +1,802 @@
...
+    var password = params.password;
+    var serverUrl = params.url;
+
+    if (serverUrl[serverUrl.length - 1] !== '/') {
+        serverUrl += '/';
+    }
+
+    var defaults = {limit:10, fetch_limit:10};
+
+    var Base64 = {
```
Comment (Spivaka) [inline Integrations/integration-ServiceNow.yml]: No need for this... you can just put
Username:
Password: 
Inside the http params (look at algosec or sandblast as ref)

**Comment by meirwah** [comment]
@gshriki plz update to resolve conflicts

**Comment by gshriki** [comment]
done. @Spivaka - your comments were addressed.
Still working on file upload so please do not merge yet

File: Integrations/integration-ServiceNow.yml [hunk 4]
```diff
@@ -0,0 +1,1321 @@
...
+    }
+
+    var SNUpdate = function(){
+        var ticket_type = getTableName();
+        var body = getBody();
+        var path = "table/"+ticket_type+"/"+args["id"];
+        var res = sendRequest('PATCH',path,null,JSON.stringify(body));
+        var md = "### ServiceNow incident updated\n";
+        md += dataToMd(command,res.obj.result);
+        return ( {'ContentsFormat': formats['json'], 'Type': entryTypes['note'], 'Contents': res.obj.result, "HumanReadable": md} );
```
Comment (slavikm) [inline Integrations/integration-ServiceNow.yml]: Cleaner to write formats.json, entryTypes.note, etc. For all the below as well...

**Pull request merged**